### PR TITLE
Add Sentry ASGI middleware

### PR DESCRIPTION
**Pull request by sarayourfriend**
Hopefully this stops the memory leak https://github.com/getsentry/sentry-python/issues/419

## Fixes

Related to #3353.

## Description

Not sure, but the memory leak could be Sentry not having the ASGI integration enabled. This PR enables it. See the linked Sentry GitHub issue for details.

## Testing Instructions

Set `SENTRY_DSN` locally and turn off `DJANGO_DEBUG_ENABLED`. Cause an exception, and see that it shows up in Sentry.

You can see the test events I sent here: https://openverse.sentry.io/issues/?environment=local&project=6107216&query=&referrer=issue-list&statsPeriod=14d

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by stacimc — APPROVED** [review]
This makes sense according to the linked issue, and I appreciate the additional comments. 🤞

File: api/conf/asgi.py [hunk 1]
```diff
@@ -15,8 +17,21 @@ def get_asgi_application():
...
+# For example, `ASGIStaticFileHandler` saves the application as
+# `self.application`. However, `SentryAsgiMiddleware` saves it as `self.app`.
+# If we relied on just referencing the pseudo-standard `application` exported
+# in this module, we'd have to choose how to resolve the base application
+# with the lifecycle handlers depending on what middleware was applied.
+# Instead, we can just take advantage of the fact that these middleware
+# all treat the passed-in application object as a singleton, so we're safe
+# to directly reference `OPENVERSE_APPLICATION`, knowing that any middleware
+# applied to it will be referencing the same application object in the end.
+OPENVERSE_APPLICATION = get_asgi_application()
```
Comment (sarayourfriend) [inline api/conf/asgi.py:32]: This is in anticipation of #3024, which is the primary reason we need the lifecycle callbacks.

**Review by AetherUnbound — APPROVED** [review]
Tested locally, works well!

**Comment by sarayourfriend** [comment]
After reading more into the Sentry Django integration, this change isn't necessary and cannot be the root cause of the memory leak. The Django integration already applies the ASGI middleware to the running application: https://github.com/getsentry/sentry-python/blob/44b0244156e1f332a8f173f337713dab99462609/sentry_sdk/integrations/django/asgi.py#L62

I've verified that the patch is working locally.

**Pull request closed without merging**